fhir-model/src/test/java/com/ibm/fhir/model/test/ResourceCoverageTest.java

```diff
@@ -0,0 +1,260 @@
+                    }
+                    
+                    if (method.getName().equals("modifierExtension") && method.toString().contains("java.util.Collection")) {
+                        List<Extension> extensions = Arrays.asList(buildBooleanTrue());
+                        builder = (Resource.Builder) method.invoke(builder, extensions);
+                    }
+                }
+                resource = builder.build();
+                
+                try (StringWriter writer = new StringWriter()) {
+                    FHIRGenerator.generator(Format.JSON).generate(resource, writer);
+                    String outJson = writer.toString();
+                    assertNotNull(outJson);
+                    assertFalse(outJson.isEmpty());
+                    
+                    try (ByteArrayInputStream in = new ByteArrayInputStream(outJson.getBytes())) {
+                        Resource resource2 = FHIRParser.parser(Format.JSON).parse(in).as(Resource.class);
+                        assertNotNull(resource2);
+                        assertTrue(true);
+                    }
+                }
+                
+                try (StringWriter writer = new StringWriter()) {
+                    FHIRGenerator.generator(Format.XML).generate(resource, writer);
+                    String outXML = writer.toString();
+                    assertNotNull(outXML);
+                    assertFalse(outXML.isEmpty());
+                    
+                    try (ByteArrayInputStream in = new ByteArrayInputStream(outXML.getBytes())) {
+                        Resource resource2 = FHIRParser.parser(Format.XML).parse(in).as(Resource.class);
+                        assertNotNull(resource2);
+                        assertTrue(true);
+                    }
+                }
+            }
+        }
+    }
+
+    public void runMethods(Object resource, Method[] methods) throws Exception {
+        for (Method method : methods) {
+            if (method.getName().startsWith("get")) {
+                method.invoke(resource);
+                assertTrue(true);
+            }
+
+            if (method.getName().equals("hashCode")) {
+                method.invoke(resource);
+                assertTrue(true);
+            }
+
+            if (method.getName().equals("equals")) {
+                method.invoke(resource, resource);
+                assertTrue(true);
+
+                String x = "BAD";
+                method.invoke(resource, x);
+                assertTrue(true);
+                
+                if (resource instanceof Resource) {
+                    Resource r = (Resource) resource; 
+                    Resource r2 = r.toBuilder().build(); 
+                    assertTrue(r.equals(r2));
+                }
+            }
+
+            if (method.getName().equals("hasChildren")) {
+                method.invoke(resource);
+                assertTrue(true);
```

Comment:
always true?

Reply:
Yes.  that's right.  it's an invocation, and asserts true as a pattern.